background.ts

```diff
@@ -747,19 +771,248 @@ function newK8sManager() {
   return mgr;
 }
 
+type validationFunc = (name: string, value: string|boolean) => string;
+
 /**
  * Implement the methods the HttpCommandServer needs to service its requests.
- * These methods should be thin wrappers around existing functionality in the rest of the backend.
- * Getters normally return strings, either JSON strings or scalars.
- * Setters normally return either `true | false`, or possibly `true | error`.
- * The `requestShutdown` is a special case that never returns.
+ * These methods do two things:
+ * 1. Verify the semantics of the parameters (the server just checks syntax).
+ * 2. Provide a thin wrapper over existing functionality in this module.
+ * Getters, on success, return status 200 and a string that may be JSON or simple.
+ * Setters, on success, return status 202, possibly with a human-readable status note.
+ * The `requestShutdown` method is a special case that never returns.
  */
 class BackgroundCommandWorker implements CommandWorkerInterface {
+  protected k8sVersions: string[] = [];
   getSettings() {
     return JSON.stringify(cfg, undefined, 2);
   }
 
-  requestShutdown() {
+  /**
+   * The core function for verifying proposed user settings.
+   * Walks the input: the user-provided object holding the new (and existing settings) against a verifier:
+   * 1. Complains about any fields in the input that aren't in the verifier
+   * 2. Recursively walks child-objects in the input and verifier
+   * 3. Calls validation functions off the verifier
+   * @param prefix - For error messages only, e.g. '', 'kubernetes.options'
+   * @param allowedSettings - The verifier
+   * @param newSettings
+   * @param errors - Builds this up as new errors are encountered, so multiple errors can be reported.
+   * @returns boolean - true if there are changes that need to be applied.
+   */
+  protected verifyProposedSettings(prefix: string,
```

Comment:
I moved the arg to the end, but decided that because it's only called with the empty string from the one exported function of the class, to not make it optional.